### PR TITLE
Fixes Explosion/Mind Blown plus multi-hit moves vs type-absorbing abilities

### PR DESCRIPTION
**Pull request by kleeenexfeu**
Fixes multi-hit moves ignoring Volt Absorb, Storm Drain, etc, after the first it.
Fixes volt absorb (and other absorbing abilities) preventing other mon on the field from being damaged by a galvanized explosion.
Same fix for mind blown, but with flash fire.
Explosion can now be used when no target are available, as stated in bulbapedia. The move will be used, the animation played, the user KOed, but it will have no effect.
Effectively removes the need of faintifabilitynotdamp battlescript command. It can freely be renamed and used for something else, and its code removed. Let me know if I have to remove it!
Basically, the only reason explosion wouldn't faint the user is if the move is not used at all (confusion, etc.) or if damp is active on the field. Same thing applied for mind blown, except for the no-targets-available case.
Fixes damp string sometimes confusing which mon's ability to display by adding a new case in the placeholder ability decoder and updating the charmap.txt.
Currently, explosion doesn't trigger eject button on a target. I didn't find any reliable source to confirm whether or not it's the right behavior. This can easily be modded by either changing the check where I left a comment in the MOVEEND_EJECT_BUTTON section, or by just swapping the order of MOVEEND_EJECT_BUTTON and the newly added MOVEEND_KO_USER in the defines.
[This](https://github.com/rh-hideout/pokeemerald-expansion/issues/1961) can be closed when this PR is merged.

It needs https://github.com/rh-hideout/pokeemerald-expansion/pull/1676 to be merged, because it accounts for parental bond.
After it is merged, the new moveend case will have to go either before or after eject button case, depending on the modern games' mechanics.

### CODE REVIEW

File: src/battle_message.c [hunk 1]
```diff
@@ -3199,6 +3199,9 @@ u32 BattleStringExpandPlaceholders(const u8 *src, u8 *dst)
             case B_TXT_ACTIVE_NAME_WITH_PREFIX: // active battlerId name with prefix
                 HANDLE_NICKNAME_STRING_CASE(gActiveBattler)
                 break;
+            case B_TXT_BATTLER_ABILITY_NAME_WITH_PREFIX:
+                HANDLE_NICKNAME_STRING_CASE(gBattlerAbility)
+            break;
```
Comment (LOuroboros) [inline src/battle_message.c:3204]:
```suggestion
                break;
```

File: src/battle_script_commands.c [hunk 2]
```diff
@@ -5266,10 +5289,35 @@ static void Cmd_moveend(void)
...
+                && (gBattleMoves[gCurrentMove].effect == EFFECT_EXPLOSION)
+                && !(gHitMarker & HITMARKER_UNABLE_TO_USE_MOVE)) // Only condition for explosion not to faint the user. Even a failed move KOes user
+            {
+                gBattleMoveDamage = gBattleMons[gBattlerAttacker].hp;
+                BattleScriptPushCursor();
+                gBattlescriptCurrInstr = BattleScript_DoRecoilNoString; // instant bar drop doesn't exist in later generations, and it saves a battlescript command
+                effect = TRUE;
+            }
+			else if (gSpecialStatuses[gBattlerAttacker].parentalBondOn != 1 // parental bonded mind blown should only hurt user on FIRST and NOT SECOND STRIKE
+																			// that's how it's written in bulbapedia at least at the time I'm writing this code (kleenexfeu)
```
Comment (LOuroboros) [inline src/battle_script_commands.c:5303]: Spaces instead of tabs.

File: src/battle_script_commands.c [hunk 3]
```diff
@@ -5266,10 +5289,35 @@ static void Cmd_moveend(void)
...
+                && IsBattlerAlive(gBattlerAttacker)
+                && (gBattleMoves[gCurrentMove].effect == EFFECT_MIND_BLOWN)
+				&& (GetBattlerAbility(gBattlerAttacker) != ABILITY_MAGIC_GUARD)
+				&& !(gMoveResultFlags & MOVE_RESULT_FAILED)
```
Comment (LOuroboros) [inline src/battle_script_commands.c:5307]: Spaces instead of tabs.

File: src/battle_script_commands.c [hunk 4]
```diff
@@ -5266,10 +5289,35 @@ static void Cmd_moveend(void)
...
+                && !(gHitMarker & HITMARKER_UNABLE_TO_USE_MOVE))
+			{
```
Comment (LOuroboros) [inline src/battle_script_commands.c:5309]: Spaces instead of tabs.

File: src/battle_script_commands.c [hunk 5]
```diff
@@ -5266,10 +5289,35 @@ static void Cmd_moveend(void)
...
+                gBattleMoveDamage = ((gBattleMons[gBattlerAttacker].maxHP + 1)/2); // damage = half max HP of user rounded up
+                BattleScriptPushCursor();
+                gBattlescriptCurrInstr = BattleScript_DoRecoilNoString;
+                effect = TRUE;
+			}
```
Comment (LOuroboros) [inline src/battle_script_commands.c:5314]: Spaces instead of tabs.

**Review by LOuroboros — CHANGES_REQUESTED** [review]

**Comment by AsparagusEduardo** [comment]
It's not compiling

**Comment by LOuroboros** [comment]
> It's not compiling

> It needs #1676 to be merged, because it accounts for parental bond.

**Pull request closed without merging**